sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -823,44 +750,80 @@ public String getEntityPath() {
+     * @throws IllegalArgumentException if {@code message.getLockToken()} is an empty value.
      */
-    public Mono<OffsetDateTime> renewMessageLock(String lockToken) {
+    public Mono<OffsetDateTime> renewMessageLock(ServiceBusReceivedMessage message) {
         if (isDisposed.get()) {
             return monoError(logger, new IllegalStateException(
                 String.format(INVALID_OPERATION_DISPOSED_RECEIVER, "renewMessageLock")));
-        } else if (Objects.isNull(lockToken)) {
-            return monoError(logger, new NullPointerException("'lockToken' cannot be null."));
-        } else if (lockToken.isEmpty()) {
-            return monoError(logger, new IllegalArgumentException("'lockToken' cannot be empty."));
+        } else if (Objects.isNull(message)) {
+            return monoError(logger, new NullPointerException("'message' cannot be null."));
+        } else if (message.getLockToken().isEmpty()) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("'message.getLockToken()' cannot be empty."));
         } else if (receiverOptions.isSessionReceiver()) {
             return monoError(logger, new IllegalStateException(
-                String.format("Cannot renew message lock [%s] for a session receiver.", lockToken)));
+                String.format("Cannot renew message lock [%s] for a session receiver.", message.getLockToken())));
         }
 
         return connectionProcessor
             .flatMap(connection -> connection.getManagementNode(entityPath, entityType))
             .flatMap(serviceBusManagementNode ->
-                serviceBusManagementNode.renewMessageLock(lockToken, getLinkName(null)))
-            .map(instant -> managementNodeLocks.addOrUpdate(lockToken, instant,
+                serviceBusManagementNode.renewMessageLock(message.getLockToken(), getLinkName(null)))
+            .map(instant -> managementNodeLocks.addOrUpdate(message.getLockToken(), instant,
                 instant.atOffset(ZoneOffset.UTC)).atOffset(ZoneOffset.UTC));
     }
 
     /**
-     * Sets the state of a session given its identifier.
+     * Starts the auto lock renewal for a message with the given lock.
+     *
+     * @param message The {@link ServiceBusReceivedMessage} to perform this operation.
+     * @param maxLockRenewalDuration Maximum duration to keep renewing the lock token.
+     *
+     * @return A lock renewal operation for the message.
+     * @throws NullPointerException if {@code message} or {@code maxLockRenewalDuration} is null.
+     * @throws IllegalStateException if the receiver is a session receiver or the receiver is disposed.
+     * @throws IllegalArgumentException if {@code message.getLockToken()} is an empty value.
+     */
+    public Mono<Void> renewMessageLock(ServiceBusReceivedMessage message, Duration maxLockRenewalDuration) {
+        if (isDisposed.get()) {
+            throw logger.logExceptionAsError(new IllegalStateException(
```

Comment:
Same question. The above uses monoError. This overload uses logger.logExceptionAsError().